src/fill_pg_plugin.hpp

```diff
@@ -20,3 +20,19 @@ class fill_pg_plugin : public appbase::plugin<fill_pg_plugin> {
   private:
     std::shared_ptr<struct fill_postgresql_plugin_impl> my;
 };
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I == sizeof...(Tp), void>::type
+variant_for_each(std::variant<Tp...>, FuncT)
+{ }
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I < sizeof...(Tp), void>::type
+variant_for_each(std::variant<Tp...> t, FuncT f)
+{
+    if (I == t.index())
+        f(I, std::get<I>(t));
+    else
+        f(I, std::variant_alternative_t<I, decltype(t)>());
```

Comment:
Why are you calling f multiple times, sometimes with default-constructed types and sometimes with the content of the variant?

Reply:
To extract all the field names to construct postgres column names.  The template is reused for both data and metadata.

Reply:
And also to fill the columns of empty variants so column name-less bulk insert will work.